packages/core-sdk/src/metadata/subgraph.ts

```diff
@@ -256,13 +260,24 @@ export async function getProductWithVariants(
 
 /** returns the productV1ItemMetadataEntities in a bundle for a given productUuid */
 // Note: it may returns several items, in case the BUNDLE contains different variants of
-// the same product (or even the same variant several times), 
-function getProductV1ItemFromBundle(bundle: BundleMetadataEntityFieldsFragment, productUuid: string):
- { productV1Item: ProductV1ItemMetadataEntity, offer: ProductV1MetadataEntityFieldsFragment["offer"] }[] {
+// the same product (or even the same variant several times),
+function getProductV1ItemFromBundle(
+  bundle: BundleMetadataEntityFieldsFragment,
+  productUuid: string
+): {
+  productV1Item: ProductV1ItemMetadataEntity;
+  offer: ProductV1MetadataEntityFieldsFragment["offer"];
+}[] {
   return bundle.items
-  .filter((item) => item.type === ItemMetadataType.ItemProductV1)
-  .filter((item) => (item as ProductV1ItemMetadataEntity).productUuid === productUuid)
-  .map((item) => {
-    return { productV1Item: (item as ProductV1ItemMetadataEntity), offer: bundle.offer as ProductV1MetadataEntityFieldsFragment["offer"] };
-  });
-}
\ No newline at end of file
+    .filter((item) => item.type === ItemMetadataType.ItemProductV1)
+    .filter(
+      (item) =>
+        (item as ProductV1ItemMetadataEntity).productUuid === productUuid
+    )
+    .map((item) => {
+      return {
+        productV1Item: item as ProductV1ItemMetadataEntity,
+        offer: bundle.offer as ProductV1MetadataEntityFieldsFragment["offer"]
+      };
+    });
+}
```

Comment:
you can get rid of some castings by adding the "is <type>" in the filter function so Typescript knows better how to narrow down the type automatically
```
  return bundle.items
    .filter(
      (item): item is ProductV1ItemMetadataEntity =>
        item.type === ItemMetadataType.ItemProductV1
    )
    .filter((item) => item.productUuid === productUuid)
    .map((item) => {
      return {
        productV1Item: item,
        offer: bundle.offer as ProductV1MetadataEntityFieldsFragment["offer"]
      };
    });
```

Reply:
Done